### PR TITLE
SPIREX-181 Pull changes from Spire Main Repository

### PR DESCRIPTION
**Pull request by willallves**

### CODE REVIEW

File: doc/plugin_agent_workloadattestor_k8s.md [hunk 1]
```diff
@@ -59,7 +59,8 @@ server name validation against the kubelet certificate.
 | k8s:pod-image-count      | The number of container images in workload's pod |
 | k8s:pod-init-image       | An Image OR ImageID of any init container in the workload's pod, [as reported by K8S](https://pkg.go.dev/k8s.io/api/core/v1#ContainerStatus). Selector value may be an image tag, such as: `docker.io/envoyproxy/envoy-alpine:v1.16.0`, or a resolved SHA256 image digest, such as `docker.io/envoyproxy/envoy-alpine@sha256:bf862e5f5eca0a73e7e538224578c5cf867ce2be91b5eaed22afc153c00363eb`|
 | k8s:pod-init-image-count | The number of init container images in workload's pod |
-
+| k8s:signature-content    | The value of the signature itself in a hash|
+| k8s:sigstore-validation   | The confirmation if the signature is valid, has value of "passed" (eg. "k8s:sigstore-validation:passed") |
```
Comment (rodrigolc) [inline doc/plugin_agent_workloadattestor_k8s.md:83]: missing other selector values for sigstore

File: test/integration/suites/k8s/conf/kind-config.yaml [hunk 2]
```diff
@@ -18,3 +18,4 @@ nodes:
   extraMounts:
   - containerPath: /etc/kubernetes/pki/admctrl
     hostPath: CONFDIR/admctrl
+
```
Comment (rodrigolc) [inline test/integration/suites/k8s/conf/kind-config.yaml]: double line ending on file ending

**Review by rodrigolc — CHANGES_REQUESTED** [review]
minor changes

File: doc/plugin_agent_workloadattestor_k8s.md (hunk 1 above)
Comment (willallves) [reply doc/plugin_agent_workloadattestor_k8s.md:83]: > missing other selector values for sigstore

added

File: README.md [hunk 3]
```diff
@@ -71,3 +71,9 @@ A third party security firm ([Cure53](https://cure53.de/)) completed a security
 ### Reporting Security Vulnerabilities
 
 If you've found a vulnerability or a potential vulnerability in SPIRE please let us know at security@spiffe.io. We'll send a confirmation email to acknowledge your report, and we'll send an additional email when we've identified the issue positively or negatively.
+
+## Sigstore workload attestor for SPIRE
+
+This repo is a fork of the [SPIRE repository](https://github.com/spiffe/spire) and enhances Kubernetes workload attestor to validate sigstore signatures.
+
+The RFC is available [here](https://docs.google.com/document/d/1YVuu7HMHnp8nx3sCPx7R2lCfjjno363s4oiPlI6axF4/edit#heading=h.ttn87ugq19sb) for reference.
```
Comment (tjamir) [inline README.md]: This is a PR. I think shouldn't have any metion that is a fork from Spire.
Instead, we should add some content to the docs descrbing the Attestor change.

File: doc/plugin_agent_workloadattestor_k8s.md [hunk 4]
```diff
@@ -42,6 +42,9 @@ server name validation against the kubelet certificate.
 | `private_key_path` | The path on disk to client key used for kubelet authentication |
 | `node_name_env` | The environment variable used to obtain the node name. Defaults to `MY_NODE_NAME`. |
 | `node_name` | The name of the node. Overrides the value obtained by the environment variable specified by `node_name_env`. |
+| `skip_signature_verification_image_list`| The list of images, described as digest hashes, that should be skipped in signature verification. |
+| `enable_allowed_subjects_list`| Enables a list of allowed subjects that are trusted and are allowed to sign container images artificats.|
+| `allowed_subjects_list`| The list of allowed subjects enabled by `enable_allowed_subjects_list` each entry represents subject e-mail. |
```
Comment (tjamir) [inline doc/plugin_agent_workloadattestor_k8s.md:47]: We missed the Rekor URL here.

File: doc/plugin_agent_workloadattestor_k8s.md [hunk 5]
```diff
@@ -117,7 +121,3 @@ WorkloadAttestor "k8s" {
   }
 }
 ```
-
-### Platform support
-
-This plugin is only supported on Unix systems.
```
Comment (tjamir) [inline doc/plugin_agent_workloadattestor_k8s.md]: In the current state of the SPIRE docs, this text is still mantained, why should we revove it?
https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md

File: doc/plugin_agent_workloadattestor_k8s.md (hunk 1 above)
Comment (tjamir) [reply doc/plugin_agent_workloadattestor_k8s.md:83]: In current version, the subject and log id are still missing.

File: support/k8s/k8s-workload-registrar/config_crd.go [hunk 6]
```diff
@@ -203,7 +205,7 @@ func (c *CRDMode) Run(ctx context.Context) error {
 			IdentityTemplate:      c.IdentityTemplate,
 			Context:               c.Context,
 			IdentityTemplateLabel: c.IdentityTemplateLabel,
-			DNSNameTemplates:      *c.DNSNameTemplates,
```
Comment (tjamir) [inline support/k8s/k8s-workload-registrar/config_crd.go:206]: Why is this line removed?

**Review by tjamir — CHANGES_REQUESTED** [review]
Some minor changes requested.
Good job!
